From d842e3222f8ce614769d1cf8227bb3b8dbf209bc Mon Sep 17 00:00:00 2001
From: Glauber Costa <glommer@redhat.com>
Date: Wed, 24 Jun 2009 14:22:57 +0100
Subject: [PATCH] use KVM_UPSTREAM for ppc.

ppc should compile with upstream qemu code, so, put these
defines in ppc specific code that references kvm functions.

We don put them in config.h in this case, since there are
files (like vl.c) that includes both kvm.h and qemu-kvm.h,
and would break compilation.

Signed-off-by: Glauber Costa <glommer@redhat.com>
Signed-off-by: Mark McLoughlin <markmc@redhat.com>
Fedora-patch: kvm-upstream-ppc.patch
---
 hw/ppc440.c            |    1 +
 hw/ppc440_bamboo.c     |    1 +
 hw/ppce500_mpc8544ds.c |    1 +
 target-ppc/helper.c    |    1 +
 4 files changed, 4 insertions(+), 0 deletions(-)

diff --git a/hw/ppc440.c b/hw/ppc440.c
index 00d82e4..164c326 100644
--- a/hw/ppc440.c
+++ b/hw/ppc440.c
@@ -18,6 +18,7 @@
 #include "ppc440.h"
 #include "ppc405.h"
 #include "sysemu.h"
+#define KVM_UPSTREAM
 #include "kvm.h"
 
 #define PPC440EP_PCI_CONFIG     0xeec00000
diff --git a/hw/ppc440_bamboo.c b/hw/ppc440_bamboo.c
index fbd447c..60ddaf4 100644
--- a/hw/ppc440_bamboo.c
+++ b/hw/ppc440_bamboo.c
@@ -21,6 +21,7 @@
 #include "boards.h"
 #include "sysemu.h"
 #include "ppc440.h"
+#define KVM_UPSTREAM
 #include "kvm.h"
 #include "kvm_ppc.h"
 #include "device_tree.h"
diff --git a/hw/ppce500_mpc8544ds.c b/hw/ppce500_mpc8544ds.c
index 8fa0383..47e35e6 100644
--- a/hw/ppce500_mpc8544ds.c
+++ b/hw/ppce500_mpc8544ds.c
@@ -16,6 +16,7 @@
 
 #include <dirent.h>
 
+#define KVM_UPSTREAM
 #include "config.h"
 #include "qemu-common.h"
 #include "net.h"
diff --git a/target-ppc/helper.c b/target-ppc/helper.c
index e02dcb0..027c8e7 100644
--- a/target-ppc/helper.c
+++ b/target-ppc/helper.c
@@ -29,6 +29,7 @@
 #include "exec-all.h"
 #include "helper_regs.h"
 #include "qemu-common.h"
+#define KVM_UPSTREAM
 #include "kvm.h"
 
 //#define DEBUG_MMU
-- 
1.6.2.5

